### PR TITLE
feat: add version 2 transaction types

### PR DESCRIPTION
**Pull request by KovacZan**
Implementation of AIP 11 for java crypto
https://github.com/ArkEcosystem/AIPs/blob/master/AIPS/aip-11.md

## What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactor
- [ ] Performance
- [ ] Tests
- [ ] Build
- [ ] Documentation
- [ ] Code style update
- [ ] Continuous Integration
- [ ] Other, please describe:

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Does this PR release a new version?

- [x] Yes
- [ ] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `develop` branch, _not_ the `master` branch
- [x] All tests are passing
- [x] New/updated tests are included

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

File: src/main/java/org/arkecosystem/crypto/transactions/builder/AbstractTransaction.java [hunk 1]
```diff
@@ -12,13 +13,18 @@
...
-        this.transaction.version = 1;
+        this.transaction.version = 2;
         this.transaction.network = Network.get().version();
+        this.transaction.typeGroup = TransactionTypeGroup.CORE;
+        this.transaction.nonce = 0;
     }
 
     public AbstractTransaction sign(String passphrase) {
+        if (this.transaction.type == TransactionType.MULTI_SIGNATURE_REGISTRATION && this.transaction.version == 2){
+             throw new UnsupportedOperationException("Version 2 MultiSignatureRegistration is not supported in java sdk");
```
Comment (faustbrian) [inline src/main/java/org/arkecosystem/crypto/transactions/builder/AbstractTransaction.java]: Why not?

Comment (KovacZan) [reply src/main/java/org/arkecosystem/crypto/transactions/builder/AbstractTransaction.java]: me and @supaiku0 didn't find well representative library for java which supports schnorr signature

File: src/main/java/org/arkecosystem/crypto/transactions/Deserializer.java [hunk 2]
```diff
@@ -25,17 +25,23 @@ public Transaction deserialize(String serialized) {
...
         transaction.network = this.buffer.get();
-        transaction.type = Types.values()[this.buffer.get()];
-        transaction.timestamp = this.buffer.getInt();
-
+        if (transaction.version == 1) {
+            transaction.type = TransactionType.values()[this.buffer.get()];
+            transaction.timestamp = this.buffer.getInt();
+        }else {
+            transaction.typeGroup = TransactionTypeGroup.values()[this.buffer.getInt()];
+            transaction.type = TransactionType.values()[this.buffer.getShort()];
```
Comment (supaiku0) [inline src/main/java/org/arkecosystem/crypto/transactions/Deserializer.java]: Not entirely sure, but doesn't this potentially cause issues when `type == 0` with `typeGroup != 1` is picked as a `Transfer` ? If Java handles it fine, nvm me.

File: src/main/java/org/arkecosystem/crypto/transactions/builder/HtlcLock.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package org.arkecosystem.crypto.transactions.builder;
+
+import org.arkecosystem.crypto.enums.HtlcLockExpirationType;
+import org.arkecosystem.crypto.enums.TransactionType;
+import org.arkecosystem.crypto.transactions.TransactionAsset;
+
+public class HtlcLock extends  AbstractTransaction {
```
Comment (supaiku0) [inline src/main/java/org/arkecosystem/crypto/transactions/builder/HtlcLock.java]: Locks can have a vendor field now.

File: src/main/java/org/arkecosystem/crypto/transactions/builder/SecondSignatureRegistration.java [hunk 4]
```diff
@@ -1,19 +1,27 @@
...
 
-    public Types getType() {
-        return Types.SECOND_SIGNATURE_REGISTRATION;
+    public TransactionType getType() {
+        return TransactionType.SECOND_SIGNATURE_REGISTRATION;
     }
 
 }
+
+class MaximumPaymentCountExceededError extends RuntimeException {
```
Comment (supaiku0) [inline src/main/java/org/arkecosystem/crypto/transactions/builder/SecondSignatureRegistration.java]: Wrong file? :hmhm:

Comment (KovacZan) [reply src/main/java/org/arkecosystem/crypto/transactions/builder/SecondSignatureRegistration.java]: yes

**Pull request merged**